test/src/multi_thread_matrix_calculation_test.cpp

```diff
@@ -482,5 +488,154 @@ TEST_F(TestMultiThreadCalculation, numSelfSubtractMatrix) {
+    executionTime = duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
+
+    // record multi-thread time in gtest
+    testing::Test::RecordProperty("MultiTime", executionTime);
+
+    // TODO this should be updated with Matrix::operator== with multi-thread
+    // make sure they are equal
+    ASSERT_TRUE(equalSingleThread(singleOutput, multiOutput, 0, singleOutput.size()));
+}
+
+TEST_F(TestMultiThreadCalculation, numDivideMatrix) {
+    auto value = generator() % MAX_VALUE, number = generator() % MAX_VALUE;
+    a = Matrix<>(squareShape, value);
+
+    auto startTime     = high_resolution_clock::now();
+    singleOutput       = number / a;
+    auto endTime       = high_resolution_clock::now();
+    auto executionTime = duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
+    // record time in gtest
+    testing::Test::RecordProperty("SingleTime", executionTime);
+
+    init(THREAD_NUM);
+
+    // the expected time in multi thread
+    testing::Test::RecordProperty("BaseTime", executionTime / (threadNum() + 1));
+
+    startTime     = high_resolution_clock::now();
+    multiOutput   = number / a;
+    endTime       = high_resolution_clock::now();
+    executionTime = duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
+
+    // record multi-thread time in gtest
+    testing::Test::RecordProperty("MultiTime", executionTime);
+
+    // TODO this should be updated with Matrix::operator== with multi-thread
+    // make sure they are equal
+    ASSERT_TRUE(equalSingleThread(singleOutput, multiOutput, 0, singleOutput.size()));
+}
+
+TEST_F(TestMultiThreadCalculation, matrixSelfSubtractNum) {
+    auto value = generator() % MAX_VALUE, number = generator() % MAX_VALUE;
+    singleOutput = multiOutput = Matrix<>(squareShape, value);
+
+    auto startTime = high_resolution_clock::now();
+    singleOutput -= number;
+    auto endTime       = high_resolution_clock::now();
+    auto executionTime = duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
+    // record time in gtest
+    testing::Test::RecordProperty("SingleTime", executionTime);
+
+    init(THREAD_NUM);
+
+    // the expected time in multi thread
+    testing::Test::RecordProperty("BaseTime", executionTime / (threadNum() + 1));
+
+    startTime = high_resolution_clock::now();
+    multiOutput -= number;
+    endTime       = high_resolution_clock::now();
+    executionTime = duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
+
+    // record multi-thread time in gtest
+    testing::Test::RecordProperty("MultiTime", executionTime);
+
+    // TODO this should be updated with Matrix::operator== with multi-thread
+    // make sure they are equal
+    ASSERT_TRUE(equalSingleThread(singleOutput, multiOutput, 0, singleOutput.size()));
+}
+
+TEST_F(TestMultiThreadCalculation, numSelfDivideMatrix) {
+    auto value = generator() % MAX_VALUE, number = generator() % MAX_VALUE;
```

Comment:
Maybe it should be value =  generator() % MAX_VALUE + 1 ,to make sure matrix isn't be none  in Divide